### PR TITLE
Remove all use of eval

### PR DESCRIPTION
**Pull request by sj26**
Let's remove all use of eval from here!

### CODE REVIEW

File: hooks/command [hunk 1]
```diff
@@ -6,43 +6,64 @@ set -ueo pipefail
...
 
-docker_compose_config_file() {
-  echo "${BUILDKITE_PLUGIN_DOCKER_COMPOSE_CONFIG:-docker-compose.yml}"
-}
-
 # Runs the docker-compose command, scoped to the project, with the given arguments
-run_docker_compose() {
-  buildkite-run "docker-compose -f $(docker_compose_config_file) -p $(docker_compose_project_name) $1"
+function run_docker_compose() {
+  command=(docker-compose)
```
Comment (toolmantim) [inline hooks/command]: Was this meant to be global? Needs a `local`

**Comment by toolmantim** [comment]
I'd be tempted to simply remove all the `buildkite-prompt` and `buildkite-run` stuff and just run the commands w/o echoing them?

**Comment by sj26** [comment]
It's really useful to see the commands when trying to debug. At the very least I think they should be output when BUILDKITE_DEBUG is true, but they're output within a squiggly section which gets hidden by default anyway.

File: hooks/command (hunk 1 above)
Comment (sj26) [inline hooks/command]: Truth!

Comment (sj26) [inline hooks/command]: Good catch! Fixed.

**Comment by toolmantim** [comment]
Nice nice. I wonder if we can provide these things to plugins? Or just provide it as a copy+paste helper that people add to their plugins?

**Comment by toolmantim** [comment]
👍 🚀

**Comment by sj26** [comment]
I don't really think we should export the `buildkite-run` etc... I'm tempted to rename them here.

I think it's up to plugin implementors to do everything beyond a basic process interface. They can definitely copy/paste useful bits! But I don't want to encourage utility libraries as I think that will quickly become a headache for all involved. :-)

**Pull request merged**